bindings/rust/s2n-tls/src/error/public.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[non_exhaustive]
+pub enum ApplicationError {
+    #[non_exhaustive]
+    CallbackExection {},
+}
```

Comment:
I don't understand the purpose of this enum. I don't see it used anywhere outside of error.rs. What's it for, and why is it public? Can you add some documentation?

Reply:
Yep ill add some documentation. The idea is that the application needs to be able to indicate failure when running custom callback code. We dont want to expose `Error`, since that is an internal impl detail. Instead we expose a non_exhaustive public version.

Here is an example of its usage: https://github.com/aws/s2n-quic/pull/1572/files#diff-749e1127d9e834de2ca8cfad3d4d5c8da3dbb89b18c783ff4c80728644caef3aR76